langwatch/src/prompt-configs/PromptConfigPanel.tsx

```diff
@@ -20,54 +31,119 @@ export function PromptConfigPanel({
   onClose,
   configId,
 }: PromptConfigPanelProps) {
+  // ---- State and hooks ----
   const { project } = useOrganizationTeamProject();
   const projectId = project?.id ?? "";
-  const { data: llmConfig } = api.llmConfigs.getByIdWithLatestVersion.useQuery(
-    {
-      id: configId,
-      projectId,
-    },
-    { enabled: !!projectId && !!configId }
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  // ---- API calls and data fetching ----
+  const {
+    mutate: invokeLLM,
+    isLoading: isExecuting,
+    data: promptExecutionResult,
+  } = useExecutePrompt();
+
+  // Fetch the LLM configuration
+  const { data: llmConfig, isLoading: isLoadingConfig } =
+    api.llmConfigs.getByIdWithLatestVersion.useQuery(
+      {
+        id: configId,
+        projectId,
+      },
+      {
+        enabled: !!projectId && !!configId,
+        staleTime: 30000, // Reduce refetches for better performance
```

Comment:
hmmm what do you mean, what was the problem you faced here? If it's already loaded then trpc will use the previous version and just refetch in the background right? I'm worried that stale time might show them an older version after saving, of if changing in another tab etc

Reply:
throwing my two cents in, for these comments it's nice to know what the exact performance reason was for the non-default setting, as this has context for the original commenter

Reply:
Yeah, good catch -- this was an AI poop that I missed. I'll remove it.